### PR TITLE
Add Datetime Library

### PR DESCRIPTION
**Pull request by RakeshJesadiya**
## Purpose of this pull request

This pull request (PR) adds a new page DateTime Library to describe the utilities provided to the date time related method.

## Affected DevDocs pages

https://devdocs.magento.com/guides/v2.3/extension-dev-guide/framework/datetime.html
https://devdocs.magento.com/guides/v2.4/extension-dev-guide/framework/datetime.html

## Links to Magento source code

https://github.com/magento/magento2/tree/2.4.1/lib/internal/Magento/Framework/Stdlib

whatsnew
Added the [DateTime library](https://devdocs.magento.com/guides/v2.4/extension-dev-guide/framework/datetime.html) topic to the _PHP Developer Guide_.

### CODE REVIEW

File: src/guides/v2.3/extension-dev-guide/framework/datetime.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+---
+group: php-developer-guide
+title: DateTime Library
+contributor_name: Rakesh Jesadiya
+contributor_link: https://github.com/RakeshJesadiya
+---
+
+## Overview
+
+This DateTime library provides utilities to work with formatting of date time. Provided methods in this library are described below.
```
Comment (keharper) [inline src/guides/v2.3/extension-dev-guide/framework/datetime.md:10]:
```suggestion
This DateTime library provides utilities to work with date and time formatting. Provided methods in this library are described below.
```

File: src/guides/v2.3/extension-dev-guide/framework/datetime.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+## Usage
+
+|Method|Description|
+|--- |--- |
+| `formatDate` | Format date to internal format |
+| `isEmptyDate` | Check whether sql date is empty |
+
+### Example
+
+The following example shows how to Format current date and time to internal format.
```
Comment (keharper) [inline src/guides/v2.3/extension-dev-guide/framework/datetime.md:21]:
```suggestion
The following example shows how to format the current date and time into the Magento internal format.
```

File: src/guides/v2.3/extension-dev-guide/framework/datetime.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+## Overview
+
+This DateTime library provides utilities to work with formatting of date time. Provided methods in this library are described below.
+
+## Usage
+
+|Method|Description|
+|--- |--- |
+| `formatDate` | Format date to internal format |
+| `isEmptyDate` | Check whether sql date is empty |
```
Comment (keharper) [inline src/guides/v2.3/extension-dev-guide/framework/datetime.md:17]: This description needs to be rewritten, but I don't know how. `sql` probably  needs to be capitalized. Is the method checking the database or somewhere else?  Maybe the description should be something like "Check whether the date has been stored in the database"

File: src/guides/v2.3/extension-dev-guide/framework/datetime.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+## Overview
+
+This DateTime library provides utilities to work with formatting of date time. Provided methods in this library are described below.
+
+## Usage
+
+|Method|Description|
+|--- |--- |
+| `formatDate` | Format date to internal format |
```
Comment (keharper) [inline src/guides/v2.3/extension-dev-guide/framework/datetime.md:16]:
```suggestion
| `formatDate` | Format the date to the Magento internal format |
```

**Review by keharper — DISMISSED** [review]
@RakeshJesadiya This PR will require technical review. Adobe/Magento will be shut down from Dec 24 to Jan 4, so it'll be a while until the PR can be processed further.

File: src/guides/v2.3/extension-dev-guide/framework/datetime.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
+---
+group: php-developer-guide
+title: DateTime Library
```
Comment (keharper) [inline src/guides/v2.3/extension-dev-guide/framework/datetime.md:3]:
```suggestion
title: DateTime library
```

File: src/guides/v2.3/extension-dev-guide/framework/datetime.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+contributor_name: Rakesh Jesadiya
+contributor_link: https://github.com/RakeshJesadiya
+---
+
+## Overview
```
Comment (keharper) [inline src/guides/v2.3/extension-dev-guide/framework/datetime.md:8]: Delete this line and the following blank link.

File: src/_data/toc/php-developer-guide.yml [hunk 7]
```diff
@@ -265,6 +265,9 @@ pages:
         - label: Array Manager
           url: /extension-dev-guide/framework/array-manager.html
 
+        - label: DateTime Library
```
Comment (keharper) [inline src/_data/toc/php-developer-guide.yml:268]:
```suggestion
        - label: DateTime library
```

**Comment by RakeshJesadiya** [comment]
@keharper I have updated the changes as per your suggestion. Thanks.

**Review by BarnyShergold — APPROVED** [review]

**Review by dobooth — APPROVED** [review]

**Comment by dobooth** [comment]
running tests

**Pull request merged**